.github/workflows/format.yml

```diff
@@ -40,13 +40,17 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+      - name: Setup pnpm
```

Comment:
1. [pnpm docs recommend using `pnpm/action-setup@v4` to setup pnpm before running `actions/setup-node@v4`](https://pnpm.io/continuous-integration#github-actions)
2. [sha pin to v4.1.0](https://github.com/pnpm/action-setup/releases/tag/v4.1.0)